torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -66,37 +67,25 @@ std::vector<at::Device> getDeviceList(const std::vector<at::Tensor>& tensors) {
   return res;
 }
 
-// Helper that lets the input ncclStreams to wait for the THC stream
+// Helper that lets the input ncclStreams to wait for the current stream
 void syncStreams(
-    THCState* thcState,
     const std::vector<at::Device>& devices,
-    std::vector<CUDAEvent>& ncclEvents,
+    std::vector<at::cuda::CUDAEvent>& ncclEvents,
     std::vector<at::cuda::CUDAStream>& ncclStreams) {
-  at::cuda::CUDAGuard gpuGuard;
   for (size_t i = 0; i < devices.size(); ++i) {
-    gpuGuard.set_index(devices[i].index());
-    auto currentThcStream =
-        THCState_getCurrentStreamOnDevice(thcState, devices[i].index());
     at::cuda::CUDAStream& ncclStream = ncclStreams[i];
-    CUDAEvent& ncclEvent = ncclEvents[i];
-
-    C10D_CUDA_CHECK(cudaEventRecord(ncclEvent.getEvent(), currentThcStream));
-    C10D_CUDA_CHECK(
-        cudaStreamWaitEvent(ncclStream.stream(), ncclEvent.getEvent(), 0));
+    at::cuda::CUDAEvent& ncclEvent = ncclEvents[i];
+    ncclEvent.record(at::cuda::getCurrentCUDAStream(devices[i].index()));
+    ncclEvent.block(ncclStream);
   }
 }
 
 } // namespace
 
 ProcessGroupNCCL::WorkNCCL::WorkNCCL(const std::vector<at::Device>& devices)
     : devices_(devices) {
-  at::cuda::CUDAGuard gpuGuard;
+  // Creates the CUDA events with DEFAULT_FLAGS = cudaEventDisableTiming
```

Comment:
This comment is incorrect -- they are created lazily upon calling record.

Reply:
Fixed.